### ISSUE TITLE
Parameterless `dotnet lambda package` fails with null path.

### ISSUE THREAD

**Issue by simonbuchan** [aws/aws-lambda-dotnet#152]
Discovered while testing #150, but this reproduces on windows-only as well, and is new in 1.7.0:

```
C:\code\tests\lambda-publish>dotnet restore
  Restoring packages for C:\code\tests\lambda-publish\lambda-publish.csproj...
  Restore completed in 65.93 ms for C:\code\tests\lambda-publish\lambda-publish.csproj.
  Generating MSBuild file C:\code\tests\lambda-publish\obj\lambda-publish.csproj.nuget.g.props.
  Restore completed in 1.18 sec for C:\code\tests\lambda-publish\lambda-publish.csproj.

C:\code\tests\lambda-publish>dotnet publish
Microsoft (R) Build Engine version 15.3.409.57025 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  lambda-publish -> C:\code\tests\lambda-publish\bin\Debug\netcoreapp1.0\lambda-publish.dll

C:\code\tests\lambda-publish>dotnet lambda package
Unknown error executing Lambda packaging: Value cannot be null.
Parameter name: paths
   at System.IO.Path.Combine(String[] paths)
   at Amazon.Lambda.Tools.LambdaPackager.CreateApplicationBundle(LambdaToolsDefaults defaults, IToolLogger logger, String workingDirectory, String projectLocation, String configuration, String targetFramework, String& publishLocation, String& zipArchivePath)
   at Amazon.Lambda.Tools.Commands.PackageCommand.<ExecuteAsync>b__22_0()
```

**Comment by normj** [comment]
I think what is going on is your project doesn't have the `aws-lambda-tools-defaults.json` where the tooling looks for the build configuration and target framework. Alternatively you could set the configuration and framework with the command line switches `--configuration` and `--framework`. 

I'm updating the tool to report a proper error message when missing the required parameters.

**Comment by normj** [comment]
I pushed out version 1.7.1 which will give better error messages.

**Comment by simonbuchan** [comment]
Is this a behavior change that defaults are required? 1.6.0 interactively prompted if flags were not provided.

**Comment by normj** [comment]
The package and package-ci commands never prompted because the assumption was they would be used in a build pipeline and I didn't want anybodys pipeline to freeze waiting for input from a user.

Although in `1.7.0` I added `--disable-interactive` to all the commands so maybe I should revert the package commands to be like the other commands to be consistent.

**Comment by simonbuchan** [comment]
1.6.0 prompted:

```
➜ PATH=/usr/bin dotnet lambda package
Enter Build Configuration: (Configuration to build with, for example Release or Debug)
Release
Enter Framework: (Target framework to compile, for example netcoreapp1.0)
netcoreapp1.0
Executing publish command
... invoking 'dotnet publish', working folder '/mnt/c/code/tests/lambda-publish/bin/Release/netcoreapp1.0/publish'
... publish: Microsoft (R) Build Engine version 15.3.409.57025 for .NET Core
... publish: Copyright (C) Microsoft Corporation. All rights reserved.
... publish: /usr/share/dotnet/sdk/2.0.0/Roslyn/Microsoft.CSharp.Core.targets(84,5): error MSB6006: "/usr/share/dotnet/sdk/2.0.0/Roslyn/RunCsc.sh" exited with code 127. [/mnt/c/code/tests/lambda-publish/lambda-publish.csproj]
Failed to create application package
```

**Comment by simonbuchan** [comment]
Not sure why it's failing like that, I was thinking probably a missed entry in `PATH`, but the same works fine in 1.7.1.

**Comment by normj** [comment]
Sorry I'm a bit confused with the last response. Is the issue fixed and we can close the issue or are you still having problems?

**Comment by simonbuchan** [comment]
Yes, I no longer get an exception stack trace in 1.7.1: I instead get `Missing required parameter: --configuration`: it's just that this is *not* the behavior of 1.6.0: that prompted for the configuration and framework. I was just checking that was an intentional change.

**Comment by normj** [comment]
That was my intention in `1.6.0` and before to not prompt because the package command was meant to be used in a CI system. That fact that it was prompting was a bug that was unintentional fixed in `1.7.0`.

**Comment by simonbuchan** [comment]
Cool, this is fine then.

The "failing like that" was referring to how 1.6.0 with an overridden `PATH` invoked the correct `dotnet`, but then `dotnet publish` failed. I've experimented a bit more, and it looks like `PATH=/usr/bin dotnet publish -c Release -f netcoreapp1.0` fails while both `PATH=/usr/bin dotnet publish` and `dotnet publish -c Release -f netcoreapp1.0` work, so that's confusing, but nothing to do with this.
